cpp/include/raft/core/mdspan_serializer.hpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
```

Comment:
What do you think about `raft/core/serialize.hpp`?

Reply:
@cjnolet Do we have serializers for other RAFT types?

Reply:
Not yet, but when we do (such as serializing scalars, key/value pairs, etc...) then we can stick them in this header.